oioioi/oi/views.py

```diff
@@ -24,6 +24,7 @@ def registration_notice_fragment(request):
         and request.user.is_authenticated
         and not is_contest_admin(request)
         and not is_participant(request)
+        and rc.can_register(request)
```

Comment:
I don't think this change in code is connected with this issue - good that you fixed that, though!

Reply:
You're right, it's only indirectly related with our issue. Without this, archived contest would still have the registration alert.